s3/pkg/meta/db/drivers/tidb/bucket.go

```diff
@@ -267,32 +293,28 @@ func (t *TidbClient) ListObjects(ctx context.Context, bucketName, marker, verIdM
 				objectNum[name] = 0
 			}
 			objectNum[name] += 1
-			marker = name
-			//filte row
-			//filte by prefix
-			hasPrefix := strings.HasPrefix(name, prefix)
-			if !hasPrefix {
-				continue
-			}
-			//filte by objectname
+			filter[common.KMarker] = name
+
 			if _, ok := objectMap[name]; !ok {
 				objectMap[name] = struct{}{}
 			} else {
 				continue
 			}
-			//filte by deletemarker
-			if deletemarker {
-				continue
-			}
+
+			// TODO: filter by deletemarker if versioning enabled
```

Comment:
please add issue to track this...any specific reason we are removing the logic to skip rows marked for delete ?

Reply:
OK, the issue is here[#713](https://github.com/opensds/multi-cloud/issues/713)